src/cljs/witan/ui/controllers/intercom.cljs

```diff
@@ -21,27 +21,39 @@
                          :widget {:activator "#IntercomDefaultWidget"}}))
     (.setInterval js/window do-update (* 1000 60 4))))
 
+(defn parse-payload [payload message]
+  (-> payload
+      (dissoc :message)
+      (assoc :message-type (or (:kixi.comms.message/type message)
+                               (:kixi.message/type message)
+                               "Unknown"))
+      (assoc :message-key  (str (or (:kixi.comms.event/key message)
+                                    (:kixi.event/type message)
+                                    (:kixi.comms.command/key message)
+                                    (:kixi.command/type message)
+                                    "n/a")))
+      (assoc :message-id   (or (:kixi.comms.event/id message)
+                               (:kixi.event/id message)
+                               (:kixi.comms.command/id message)
+                               (:kixi.command/id message)
+                               "n/a"))))
+
+
+(defn publish-to-intercom [event-label name activity payload message]
+  (.Intercom js/window
+             event-label
+             (name activity)
+             (clj->js (parse-payload payload message))))
+
+(defn on-panic-event
+  [{:keys [args]}]
+  (let [{:keys [activity message] :as payload} args]
+    (publish-to-intercom "panicEvent" name activity payload message)))
```

Comment:
I don't understand the difference between panicEvent and trackEvent, but maybe @acron0 can explain.